src/net/http.rs

```diff
@@ -1143,28 +1177,120 @@ impl HttpRequestType {
     }
     
-    fn parse_getneighbors<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, _regex: &Regex, _fd: &mut R) -> Result<HttpRequestType, net_error> {
+    fn parse_getneighbors<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, _regex: &Captures, _fd: &mut R) -> Result<HttpRequestType, net_error> {
         if preamble.get_content_length() != 0 {
             return Err(net_error::DeserializeError("Invalid Http request: expected 0-length body for GetNeighbors".to_string()));
         }
 
         Ok(HttpRequestType::GetNeighbors(HttpRequestMetadata::from_preamble(preamble)))
     }
 
-    fn parse_getblock<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, regex: &Regex, _fd: &mut R) -> Result<HttpRequestType, net_error> {
+    fn parse_get_transfer_cost<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, _regex: &Captures, _fd: &mut R) -> Result<HttpRequestType, net_error> {
+        if preamble.get_content_length() != 0 {
+            return Err(net_error::DeserializeError("Invalid Http request: expected 0-length body for GetTransferCost".to_string()));
+        }
+
+        Ok(HttpRequestType::GetTransferCost(HttpRequestMetadata::from_preamble(preamble)))
+    }
+
+    fn parse_get_account<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, captures: &Captures, _fd: &mut R) -> Result<HttpRequestType, net_error> {
+        if preamble.get_content_length() != 0 {
+            return Err(net_error::DeserializeError("Invalid Http request: expected 0-length body for GetAccount".to_string()));
+        }
+
+        let principal = PrincipalData::parse(&captures["principal"])
+            .map_err(|_e| net_error::DeserializeError("Failed to parse account principal".into()))?;
+
+        Ok(HttpRequestType::GetAccount(HttpRequestMetadata::from_preamble(preamble), principal))
+    }
+
+    fn parse_get_map_entry<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, captures: &Captures, fd: &mut R) -> Result<HttpRequestType, net_error> {
+        let content_len = preamble.get_content_length();
+        if !(content_len > 0 && content_len < (MAX_VALUE_SIZE * 4)) {
+            return Err(net_error::DeserializeError("Invalid Http request: invalid body length for GetMapEntry".to_string()));
+        }
+
+        let contract_addr =  StacksAddress::from_string(&captures["address"])
+            .ok_or_else(|| net_error::DeserializeError("Failed to parse contract address".into()))?;
+        let contract_name = ContractName::try_from(captures["contract"].to_string())
+            .map_err(|_e| net_error::DeserializeError("Failed to parse contract name".into()))?;
+        let map_name = ClarityName::try_from(captures["map"].to_string())
+            .map_err(|_e| net_error::DeserializeError("Failed to parse contract name".into()))?;
+
+        let value_hex: String = serde_json::from_reader(fd)
```

Comment:
What's the maximum length of `value_hex`?  Knowing this will help determine whether or not to include the `key` in the URL string (which would have the effect of bounding it to fit inside a HTTP preamble, which is 4096 bytes right now).

Reply:
Nm, this can be too big to fit inside a URL.

Reply:
It is at least 1MB inflated 2x (because of the hex string). But the serialization of Clarity values does have some overhead as well, which I don’t want to set an exact limit on, because I’d much rather err on having some leniency in accepting data for now (since the serialization code will abort too large of a value anyways) rather than having a situation where some corner case value’s serialization is actually larger than whatever limit we impose.

Either way, 2MB is way too big for a path string.